app/furniture/marketplace/carts/_footer.html.erb

```diff
@@ -8,8 +8,9 @@
   <tr>
     <td class="text-left px-3 py-3.5 text-right" colspan="8">
       <%= render "marketplace/carts/total", cart: cart %>
-      <%= button_to("Checkout", cart.location(child: :checkout), data: { turbo: false }) %>
+      <%- if cart.delivery_address.present? %>
+        <%= button_to("Checkout", cart.location(child: :checkout), data: { turbo: false }) %>
+      <%- end %>
```

Comment:
this seems to make the checkout button hidden if the user never inputted a `cart.delivery_address`. Customers might worry when they don't see the checkout button.

Why not just have the `Checkout` disabled if `cart.delivery_address` does not exist or is empty string? Or always keep `Checkout` enabled,  but if they submit the form without the `cart.delivery_address`, do not let the form submit and inform them that `cart.delivery_address` is required.

Reply:
`disabled` is better than hidden for sure! I considered making it so if it wasn't set, the `CheckoutsController#show` action gave them an affordance for adding the `Cart#delivery_address` but I decided against it because it was more code.

I also considered putting the entire cart *except* for the `delivery_address` behind a `if/else` block; but that seemed excessive.

Reply:
@KellyAH - Do you have a preference between:

1. Disabling the `checkout` button unless the `delivery_address` is set, _or_
2. Hiding the `cart` until it has a `delivery_address` 

Part of me thinks option 2 is actually worth doing; since it serves as a [forcing function](https://www.interaction-design.org/literature/book/the-glossary-of-human-computer-interaction/forcing-functions) and reduces the amount of elements on the page significantly...

Reply:
wouldn't 2️⃣  `hiding the cart`, freak the customer out since they can't see their cart items. thus they don't know what is in their cart.

IMHO 1️⃣ makes it clear that some other required action is needed to be allowed to proceed to the checkout flow.

Reply:
I think we can *presume* a lot about what would or would not freak a customer out. It's not terribly unusual for clicking a link to navigate to a different page; so a customer freaking out because when they clicked a link they went somewhere else would surprise me.

Reply:
My hesitation on 2️⃣  is that I just couldn't visualize how it would look in the UI/UX. 

If the flow looks fine with the 2️⃣ change and is something you as a customer would be comfy with. I say do 2️⃣ 👍 😄

Reply:
Here is 2️⃣'s flow! I think it's pretty decent (but it needs better text! lol)

https://user-images.githubusercontent.com/50284/222829834-ff257c74-35e8-488c-a5cf-65e0e4e2fce1.mp4